cli/lib/tasks/download.js

```diff
@@ -196,32 +196,27 @@ const downloadFromUrl = ({ url, downloadDestination, progress, ca }) => {
       downloadDestination,
     })
 
-    let redirectVersion
+    if (ca) debug('using custom CA details from npm config')
 
+    let redirectVersion
+    let redirectUrl
     const reqOptions = {
-      url,
-      proxy,
+      uri: url,
+      ...(proxy ? { proxy } : {}),
+      ...(ca ? { ca } : {}),
```

Comment:
Do we need to nest the ca under agentOptions?

```suggestion
      ...(ca ? { agentOptions: { ca }} : {}),
```

Reply:
good catch